PWGJE/Tasks/PhotonIsolationQA.cxx

```diff
@@ -0,0 +1,364 @@
+
+    MC_Info.add("hEvsNumTracks", "Energy of cluster vs matched tracks", o2HistType::kTH2F, {{Energy_Axis}, {Num_Track_Axis}});
+    MC_Info.add("hTrackPt", "Pt of matched track", o2HistType::kTH1F, {Pt_Axis});
+    MC_Info.add("hTrackEta", "Eta of matched track", o2HistType::kTH1F, {Eta_Axis});
+    MC_Info.add("hTrackPhi", "Phi of matched track", o2HistType::kTH1F, {Phi_Axis});
+    MC_Info.add("hRatioClusterETrackP", "Ratio between energy of cluster and P of tracks", o2HistType::kTH1F, {RatioP_Axis});
+    MC_Info.add("hRatioClusterETrackPrackPt", "Ratio between E and P vs Pt of matched tracks", o2HistType::kTH2F, {{RatioP_Axis}, {Pt_Axis}});
+
+    MC_Info.add("hEvsPtIso", "Pt_Iso", o2HistType::kTH2F, {{Energy_Axis}, {PtIso_Axis}});
+    MC_Info.add("hRho_Perpen_Cone", "Density of perpendicular cone", o2HistType::kTH1F, {Rho_Axis});
+    MC_Info.add("hShowerShape", "Shower shape", o2HistType::kTH2F, {{Shower_Shape_Long_Axis}, {Shower_Shape_Short_Axis}});
+    MC_Info.add("hSigmaLongvsPtIso", "Long shower shape vs Pt_Iso", o2HistType::kTH2F, {{Shower_Shape_Long_Axis}, {PtIso_Axis}});
+    MC_Info.add("hABCDControlRegion", "Yield Control Regions", o2HistType::kTH2F, {{ABCD_Axis}, {Energy_Axis}});
+    MC_Info.add("hMcParticlesToCluster", "Energy of particles linked to mc cluster", o2HistType::kTH1F, {{100, 0, 20}});
+    MC_Info.add("hMcParticleStatusCode", "status code of mc particle linked to mc cluster", o2HistType::kTH1F, {{100, 0, 100}});
+
+    std::vector<std::string> bin_names = {"A", "B", "C", "D", "True Bckgr A"};
+    for (int i = 0; i < bin_names.size(); i++) {
+      MC_Info.get<TH2>(HIST("hABCDControlRegion"))->GetXaxis()->SetBinLabel(i + 1, bin_names[i].c_str());
+      Data_Info.get<TH2>(HIST("hABCDControlRegion"))->GetXaxis()->SetBinLabel(i + 1, bin_names[i].c_str());
+    }
+  }
+
+  // boolian returns true if a track is matched to the cluster with E_Cluster/P_track < 1.75. Otherwise returns false
+  bool track_matching(const auto& cluster, o2::aod::EMCALMatchedTracks const& matched_tracks)
+  {
+    for (const auto& match : matched_tracks) {
+      double dEta = abs(match.track_as<myGlobTracks>().trackEtaEmcal() - cluster.eta());
+      double dPhi = match.track_as<myGlobTracks>().trackPhiEmcal() - cluster.phi();
+      if (dPhi < 0) {
+        dPhi = 2. * M_PI - dPhi;
+      }
+      double distance = sqrt(pow(dEta, 2) + pow(dPhi, 2));
+      if (distance <= TrackMatchingRadius) {
+        double abs_p = abs(match.track_as<myGlobTracks>().p());
+        if ((cluster.energy() / abs_p) < 1.75) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  // sums the pt of all tracks around the cluster within a radius of R = 0.4
+  double sum_Pt_tracks_in_cone(const auto& cluster, myGlobTracks const& tracks, double Cone_Radius = 0.4)
+  {
+    double sum_Pt = 0.;
+    for (const auto& track : tracks) {
+      double dphi = cluster.phi() - track.phi();
```

Comment:
dphi calculation is I think not fully correct? e.g. for phi from 0 to 2pi, if cluster phi would theoretically be close to 2pi and track phi very close to 0, then you would get a very large dPhi distance even though cluster and track are close. Double check definitions throughout. Maybe I am wrong

Reply:
You are correct. I have fixed that bug in the PR that is now pending (https://github.com/AliceO2Group/O2Physics/pull/6689). In the pending PR i also fixed the track matching (i think). The emcal-correction task will now fully handle the tracks matched to a cluster